Significant updates made to internals of csv-express

lpage opened the pull request:
While code changes I've made suit my own purposes, I did not change the public API (as compared to v1.2.3 of your fork of the project). All existing unit tests still pass on a variety of node and express versions, and I was able to achieve a small performance improvement over v1.2.3. I've bumped the version of my fork of this project to v.1.2.9 given the number of changes I made. Please review ./history.md for more details. The docs directory I added is for the JSDoc output I created for the ./lib/csv-express.js file. Thanks.

lpage committed:
refactor lib/csv-express.js; add more test scripts; add istanbul and jsdoc deps; update history.md

lpage committed:
update history.md

lpage committed:
update history.md again

jczaplew commented:
Hi @lpage, thanks for the pull request. Given that your refactor is a complete rewrite of `csv-express` and adds tens of thousands of lines of code, I think this is better off published as a separate package.

The main code of `csv-express` is a little over 100 lines of Javascript, and I would like to keep this fork of it as lightweight as possible.

lpage commented:
Hi John,

The pull request does seem like it has a lot of changes, but that's only
because I added documentation files (generated by JSDoc) to the pull
request; you can ignore these doc files if you find no value in merging
them into your fork. The only significant code changes I made were in
./lib/csv-express.js. If you compare what's in your fork of the project for
that file compared to what I have, you'll see that I haven't added much in
terms of actual lines of code (especially if you disregard the additional
JSDoc comments I added to the file). All in all, both your version and mine
have pretty close to the same lines of code. I did a code-line count to
compare yours to mine. Excluding comments, you have 70 lines of source code
in your version of csv-express.js, whereas I have 99 lines. You'll see that
my version leans more in the direction of a functional style of code vs.
the imperative style of the original.

If you'd like, I can issue another pull request, excluding all of the doc
files. I can admit that I almost excluded all the doc files in my original
pull request, thinking that it might seem like a lot to merge in ;-)

Thanks for getting back to me.

Cheers,

Lawrence

On Tue, May 16, 2017 at 7:41 AM, John J Czaplewski <notifications@github.com
> wrote:

> Hi @lpage <https://github.com/lpage>, thanks for the pull request. Given
> that your refactor is a complete rewrite of csv-express and adds tens of
> thousands of lines of code, I think this is better off published as a
> separate package.
>
> The main code of csv-express is a little over 100 lines of Javascript,
> and I would like to keep this fork of it as lightweight as possible.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jczaplew/csv-express/pull/8#issuecomment-301804330>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABcpUHp2YjwmDcVdDFcxFkD-f2FC4bHIks5r6bW3gaJpZM4Nb3yn>
> .
>
